numba/cuda/tests/cudapy/test_operator.py

```diff
@@ -105,6 +137,118 @@ def test_floordiv(self):
         self.operator_template(operator.floordiv)
 
     @skip_unless_cc_53
+    def test_fp16_binary(self):
+        functions = (simple_fp16add, simple_fp16sub, simple_fp16mul)
+        ops = (operator.add, operator.sub, operator.mul)
+
+        for fn, op in zip(functions, ops):
+            with self.subTest(op=op):
+                kernel = cuda.jit("void(f2[:], f2, f2)")(fn)
+
+                expected = np.zeros(1, dtype=np.float16)
+                got = np.zeros(1, dtype=np.float16)
+                arg1 = np.random.random(1).astype(np.float16)
+                arg2 = np.random.random(1).astype(np.float16)
+
+                kernel[1, 1](got, arg1[0], arg2[0])
+                expected = op(arg1, arg2)
+                np.testing.assert_allclose(got[0], expected)
```

Comment:
This compares the scalar first element of `got` with the array `expected`. It would be simpler and clearer to just compare the two arrays.
```suggestion
                np.testing.assert_allclose(got, expected)
```